### ISSUE TITLE
Regression: pages with list_in_left_nav=False are showing up in left nav

### ISSUE THREAD

**Issue by pdpinch** [mitodl/ocw-to-hugo#281]
See https://ocwnext.odl.mit.edu/courses/11-405-just-money-banking-as-if-society-mattered-spring-2021/

The JSON at s3://open-learning-course-data-production/11-405-just-money-banking-as-if-society-mattered-spring-2021
 contains:

```
    {
      "list_in_left_nav": false, 
      "file_location": "42302a80a779eca6385816bd7c7a6bf1_syllabus.html", 
      "type": "CourseSection", 
      "uid": "42302a80a779eca6385816bd7c7a6bf1", 
      "title": "Syllabus", 
      "url": "/courses/urban-studies-and-planning/11-405-just-money-banking-as-if-society-mattered-spring-2021/syllabus", 
      "text": "", 
      "is_image_gallery": false, 
      "order_index": null, 
      "bottomtext": "", 
      "short_page_title": "Syllabus", 
      "short_url": "syllabus", 
      "parent_uid": "582eb8f5f39dfebe0aef874da6a2623e", 
      "is_media_gallery": false, 
      "description": ""
    }
```

**Comment by pdpinch** [comment]
While this is technically a regression, we may also need to reconsider the feature. It seems that on the legacy site that if a page has `"excludeFromNav": false` **_and_** `"always_publish": false` the page shouldn't be published at all.

**Comment by noisecapella** [comment]
@pdpinch What is the expected behavior here? I'm not familiar with what `list_in_left_nav` does

**Comment by pdpinch** [comment]
Sorry George, the history of issues and PRs on this are not clear. I updated the description and I hope it makes sense now.

**Comment by gumaerc** [comment]
@pdpinch FYI the reason for the link in your example showing up when `list_in_left_nav` is set to false is because of this:

```
  if (inRootNav || listInLeftNav) {
    courseSectionFrontMatter["menu"] = {
      leftnav: {
        identifier: pageId,
        name:       shortTitle || "",
        weight:     menuIndex
      }
    }
    if (parentId) {
      courseSectionFrontMatter["menu"]["leftnav"]["parent"] = parentId
    }
  }
```

Since Syllabus is in the root of the nav, it's shown regardless.  It looks like I had come to that conclusion here: https://github.com/mitodl/hugo-course-publisher/issues/181.  It seems at that point, `list_in_left_nav` was not being set at all for root level items so it was assumed that they were to be shown regardless.

**Comment by noisecapella** [comment]
Yes, as Carey said `list_in_left_nav` has the correct values for `11-405-just-money-banking-as-if-society-mattered-spring-2021` but it doesn't have the correct values for `20-020-introduction-to-biological-engineering-design-spring-2009` which is the course linked in https://github.com/mitodl/hugo-course-publisher/issues/181, so I don't think we can use it for the sidebar

**Comment by noisecapella** [comment]
@pdpinch For example, the Syllabus page is listed in the sidebar:
![Screenshot from 2021-06-23 09-34-55](https://user-images.githubusercontent.com/863262/123106177-6a1b5300-d406-11eb-8696-eb81b53d6df9.png)

But it is `false` in the parsed JSON:
![Screenshot from 2021-06-23 09-35-39](https://user-images.githubusercontent.com/863262/123106626-c5e5dc00-d406-11eb-81fc-16ca712d602a.png)

**Comment by pdpinch** [comment]
Here's another example where 

```
"list_in_left_nav": false,
```

but the item still appears in the left nav. 

OCW Next: https://ocwnext.odl.mit.edu/courses/6-036-introduction-to-machine-learning-fall-2020/
Legacy: https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-036-introduction-to-machine-learning-fall-2020/

**Comment by pdpinch** [comment]
After investigating this further, it looks like we maybe should be using a different key from the raw Plone JSON, `excludeFromNav`, instead of `list_in_left_nav`
